stable/admin/templates/_sidecar.tpl

```diff
@@ -0,0 +1,86 @@
+{{/*
+Return the proper nuocollector image name
+*/}}
+{{- define "nuocollector.image" -}}
+{{- $registryName := $.Values.nuocollector.image.registry -}}
+{{- $repositoryName := $.Values.nuocollector.image.repository -}}
+{{- $tag := $.Values.nuocollector.image.tag | toString -}}
+{{/*
+Helm 2.11 supports the assignment of a value to a variable defined in a different scope,
+but Helm 2.9 and 2.10 doesn't support it, so we need to implement this if-else logic.
+Also, we can't use a single if because lazy evaluation is not an option
+*/}}
+{{- if .Values.global }}
+    {{- if .Values.global.imageRegistry }}
+        {{- printf "%s/%s:%s" .Values.global.imageRegistry $repositoryName $tag -}}
+    {{- else -}}
+        {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
+    {{- end -}}
+{{- else -}}
+    {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the proper nuocollector config watcher image name
+*/}}
+{{- define "nuocollector.watcher" -}}
+{{- $registryName := .Values.nuocollector.watcher.registry -}}
+{{- $repositoryName := .Values.nuocollector.watcher.repository -}}
+{{- $tag := .Values.nuocollector.watcher.tag | toString -}}
+{{/*
+Helm 2.11 supports the assignment of a value to a variable defined in a different scope,
+but Helm 2.9 and 2.10 doesn't support it, so we need to implement this if-else logic.
+Also, we can't use a single if because lazy evaluation is not an option
+*/}}
+{{- if .Values.global }}
+    {{- if .Values.global.imageRegistry }}
+        {{- printf "%s/%s:%s" .Values.global.imageRegistry $repositoryName $tag -}}
+    {{- else -}}
+        {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
+    {{- end -}}
+{{- else -}}
+    {{- printf "%s/%s:%s" $registryName $repositoryName $tag -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "nuodb.sidecar" -}}
+{{- if $.Values.nuocollector }}
```

Comment:
Does this '{{ $.Values.nuocollector }}' have different behaviour to '{{ .Values.nuocollector }}' ?

If not, then could we please have all value references be consistent?

Currently, all references to values from the values.yaml file are in the form {{ .Values... }}
and only variables have the '$' prefix - since they must.

Reply:
@NikTJ777 The `$` is mapped to the root scope when template execution begins and it does not change during template execution. It is required when root or parent scope needs to be accessed within a scoped block. In this specific case, I don't think it's needed. The PR is merged already so we may want to address this in subsequent change.